js/libs/builder.js

```diff
@@ -292,49 +292,35 @@ var app = new Vue({
 
         var operation;
 
-        if ($vm.settings.dataStore.indexOf('dataSource') > -1 || $vm.settings.dataSourceId) {
+        if ($vm.settings.dataStore.indexOf('dataSource') > -1 && $vm.settings.dataSourceId) {
           var newHook = {
-            widgetInstanceId: $vm.settings.id,
+            widgetInstanceId: widgetId,
             runOn: ['insert'],
             type: 'email',
-            payload: $vm.emailTemplateAdd
+            payload: JSON.parse(JSON.stringify($vm.emailTemplateAdd))
           };
 
           operation = Fliplet.DataSources.getById($vm.settings.dataSourceId).then(function(dataSource) {
-            var currentHook;
+            // Remove old hook
+            dataSource.hooks = _.reject(dataSource.hooks, { type: 'email', widgetInstanceId: widgetId });
```

Comment:
Line 305: This can't be like this, because there might be a hook for the Edit template, and will match the same `type` and `widgetId` and you don't want to delete it.
Use the `runOn` key instead.

Reply:
ah true, thanks